src/lib/vina.cpp

```diff
@@ -285,39 +285,35 @@ void Vina::set_forcefield() {
 	m_scoring_function = scoring_function;
 }
 
-void Vina::set_vina_box(double center_x, double center_y, double center_z, double size_x, double size_y, double size_z, double granularity, bool force_even_voxels) {
-	// Setup the search box
-	if (size_x <= 0 || size_y <= 0 || size_z <= 0) {
-		std::cerr << "ERROR: Search space dimensions should be positive.\n";
-		exit(EXIT_FAILURE);
-	} else if (size_x * size_y * size_z > 27e3) {
-		std::cerr << "WARNING: The search space volume > 27000 Angstrom^3 (See FAQ)\n";
-	}
+std::vector<double> Vina::grid_dimensions_from_ligand(double buffer_size) {
+	std::vector<double> box_dimensions(6, 0);
+	std::vector<double> box_center(3, 0);
+	std::vector<double> max_distance(3, 0);
 
-	grid_dims gd;
-	vec span(size_x, size_y, size_z);
-	vec center(center_x, center_y, center_z);
+	// The center of the ligand will be the center of the box
+	box_center = m_model.center();
 
-	VINA_FOR_IN(i, gd) {
-		gd[i].n_voxels = sz(std::ceil(span[i] / granularity));
+	// Get the furthest atom coordinates from the center in each dimensions
+	VINA_FOR(i, m_model.num_movable_atoms()) {
+		const vec& atom_coords = m_model.get_coords(i);
 
-		// If odd n_voxels increment by 1
-		if (force_even_voxels && (gd[i].n_voxels % 2 == 1))
-			// because sample points (npts) == n_voxels + 1
-			gd[i].n_voxels += 1;
+		VINA_FOR_IN(j, atom_coords) {
+			double distance = std::sqrt(std::pow((box_center[j] - atom_coords[j]), 2));
```

Comment:
Since this is done for each coordinate, why not just `double distance = fabs(box_center[j]-atom_coords[j]);` ;-)

Reply:
True... --'